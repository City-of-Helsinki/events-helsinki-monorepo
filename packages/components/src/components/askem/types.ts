import type { RnsData } from '../../types';
import type Askem from './Askem';

export type AskemBaseConfig = {
  disabled?: boolean;
  scriptUrl?: string;
  consentGiven?: boolean;
};

export type AskemConfigs = AskemBaseConfig & RnsData;

export interface AskemInstance {
  disabled: boolean;
  setRnsConfigValue: Askem['setRnsConfigValue'];
  consentGiven: boolean;
}
