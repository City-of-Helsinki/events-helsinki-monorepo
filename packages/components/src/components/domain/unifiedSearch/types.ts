import type {
  OpeningHours,
  OpeningHoursTimes,
  SearchListQueryVariables,
} from '../../../types';
import type { OrderByType, OrderDirType } from './unifiedSearchConstants';

export type UnifiedSearchParameters = {
  q?: string[];
  administrativeDivisionIds?: string[];
  ontologyTreeIds?: number[];
  ontologyWordIds?: number[];
  after?: string;
  first?: number;
  ontology?: string;
  isOpenNow?: boolean;
  openAt?: Date;
  orderBy?: OrderByType;
  orderDir?: OrderDirType;
};
export type UnifiedSearchVariables = SearchListQueryVariables;
export type UnifiedSearchOpeningHours = OpeningHours;
export type UnifiedSearchOpeningHoursTimes = OpeningHoursTimes;

export type TransitionOptions = {
  shallow?: boolean;
  locale?: string | false;
  scroll?: boolean;
};

export type ModifyFilters = (
  search: Partial<UnifiedSearchParameters>,
  transitionOptions?: TransitionOptions
) => void;
