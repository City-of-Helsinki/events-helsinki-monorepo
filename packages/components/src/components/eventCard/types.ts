import type { EventFields } from '../../types';

export type EventCardProps = {
  event: EventFields;
  eventUrl: string;
};

export type LargeEventCardProps = EventCardProps & {
  showEnrolmentStatusInCardDetails?: boolean;
};
