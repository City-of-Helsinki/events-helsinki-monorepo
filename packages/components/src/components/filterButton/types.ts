export type FilterType =
  | 'category'
  | 'hobbyType'
  | 'sportsCategory'
  | 'date'
  | 'dateType'
  | 'division'
  | 'place'
  | 'publisher'
  | 'target'
  | 'text'
  | 'minAge'
  | 'maxAge'
  | 'exactAge';
