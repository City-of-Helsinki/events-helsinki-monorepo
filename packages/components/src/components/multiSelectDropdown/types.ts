import type { Option } from '../../types/types';

export interface MultiselectDropdownProps {
  checkboxName: string;
  icon: React.ReactElement;
  inputPlaceholder?: string;
  inputValue?: string;
  name: string;
  onChange: (values: string[]) => void;
  options: Option[];
  fixedOptions?: Option[];
  renderOptionText?: (optionValue: string) => React.ReactChild;
  selectAllText?: string;
  setInputValue?: (newVal: string) => void;
  showSearch?: boolean;
  showSelectAll?: boolean;
  title: string;
  value: string[];
  className?: string;
  helpText?: string;
  filterByInput?: boolean;
}
