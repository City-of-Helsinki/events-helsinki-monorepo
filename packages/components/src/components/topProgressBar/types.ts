type LoggerFunction = (
  message?: string,
  ...optionalParameters: unknown[]
) => void;

export type Logger = {
  debug?: LoggerFunction;
  info?: LoggerFunction;
  warn?: LoggerFunction;
  error?: LoggerFunction;
};
