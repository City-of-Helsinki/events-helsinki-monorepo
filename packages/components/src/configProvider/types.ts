import type { ApolloClient, NormalizedCacheObject } from '@apollo/client';
import type { TFunction } from 'next-i18next';
import type { NextRouter } from 'next/router';

export interface Config {
  router?: NextRouter; // TODO: Support the react-router
  apolloClient?: ApolloClient<NormalizedCacheObject>;
  t: TFunction;
  /* eslint-disable @typescript-eslint/naming-convention */
  components: {
    A: (props: React.AnchorHTMLAttributes<HTMLAnchorElement>) => JSX.Element;
    Link?: (
      props: React.AnchorHTMLAttributes<HTMLAnchorElement>
    ) => JSX.Element;
  };
  /* eslint-enable @typescript-eslint/naming-convention */
}
