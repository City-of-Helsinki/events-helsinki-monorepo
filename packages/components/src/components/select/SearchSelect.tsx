import classNames from 'classnames';
import styles from './searchSelect.module.scss';
import Select from './Select';

/** Select-component with Search form styles. */
export default function SearchSelect({
  // FIXME: While the HDS-select is bugged, the value property cannot be used, or the value control fails.
  // While the bug exists, we are using the defaultValue instead...
  // value,
  defaultValue: value,
  ...rest
}: Parameters<typeof Select>[0]) {
  // The value can be an array of options, a single option, null or undefined.
  // The option value can also be empty!
  const hasValue = Array.isArray(value) ? value.length : !!value?.value;
  return (
    <Select
      {...rest}
      value={undefined}
      defaultValue={value}
      className={classNames([
        styles.searchSelect,
        {
          [styles.isSelected]: hasValue,
        },
      ])}
    />
  );
}
