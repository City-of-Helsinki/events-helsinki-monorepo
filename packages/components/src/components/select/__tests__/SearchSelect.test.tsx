import { screen, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import type { Option } from '../../../types';
import { AccessibilityProfile } from '../../../types';

import SearchSelect from '../SearchSelect';
// eslint-disable-next-line import/no-unresolved

const emptyOption: Option = { text: '', value: '' };
const accessibilityProfileOptions = [
  emptyOption,
  ...Object.values(AccessibilityProfile).map(
    (accessibility) => ({ text: accessibility, value: accessibility } as Option)
  ),
];

describe('SearchSelect', () => {
  it('renders properly', () => {
    const { container } = render(
      <SearchSelect
        id="accessibilityProfile"
        label="label"
        placeholder="placeholder"
        options={accessibilityProfileOptions}
      />
    );
    expect(
      screen.getByRole('button', {
        name: /label/i,
      })
    ).toBeInTheDocument();
    expect(screen.getByText(/placeholder/i)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("lists the values when it's clicked", async () => {
    render(
      <SearchSelect
        id="accessibilityProfile"
        label="label"
        placeholder="placeholder"
        options={accessibilityProfileOptions}
      />
    );
    expect(await screen.findByText(/placeholder/i)).toBeInTheDocument();
    // No list items are shown
    accessibilityProfileOptions
      .slice(1)
      .forEach((option) =>
        expect(screen.queryByText(option.text)).not.toBeInTheDocument()
      );
    await userEvent.click(
      screen.getByRole('button', {
        name: /label/i,
      })
    );
    // All list items are shown
    accessibilityProfileOptions
      .slice(1)
      .forEach(async (option) =>
        expect(await screen.findByText(option.text)).toBeInTheDocument()
      );
    // Select one
    await userEvent.click(
      //   screen.getByRole('option', {
      //     name: /rollator/i,
      //   })
      screen.getByText(/rollator/i)
    );
    // The placeholder text is replaced with the selected option
    expect(screen.queryByText(/placeholder/i)).not.toBeInTheDocument();
    expect(screen.getByText(/rollator/i)).toBeInTheDocument();
  });

  it.each(['value', 'defaultValue'])(
    'does not allow array as a %s',
    (field) => {
      expect(() =>
        render(<SearchSelect label="label" options={[]} {...{ [field]: [] }} />)
      ).toThrow(/must be singletons/);
    }
  );

  it('does not allow multiselect', () => {
    expect(() =>
      // @ts-ignore
      render(<SearchSelect label="label" options={[]} multiselect />)
    ).toThrow(/does not support multiselect feature/);
  });
});
