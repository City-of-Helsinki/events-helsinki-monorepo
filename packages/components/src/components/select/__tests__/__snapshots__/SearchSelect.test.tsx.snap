// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchSelect > successful cases > renders properly 1`] = `
<div>
  <div
    class="Select-module_root__Ka5uO _select_12d8cc _searchSelect_481aa4"
  >
    <label
      class="FieldLabel-module_label__1zrXK "
      for="accessibilityProfile-toggle-button"
      id="accessibilityProfile-label"
    >
      label
    </label>
    <div
      class="Select-module_wrapper__1WQXs"
    >
      <button
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="accessibilityProfile-label accessibilityProfile-toggle-button"
        aria-owns="accessibilityProfile-menu"
        class="Select-module_button__1aIsm Select-module_placeholder__21c0j"
        id="accessibilityProfile-toggle-button"
        type="button"
      >
        <span
          class="Select-module_buttonLabel__1fqu5"
        >
          placeholder
        </span>
        <svg
          aria-hidden="true"
          aria-label="angle-down"
          class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik Select-module_angleIcon__2-9AD"
          role="img"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M12 13.5L17 8.5L18.5 10L12 16.5L5.5 10L7 8.5L12 13.5Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </button>
      <ul
        aria-labelledby="accessibilityProfile-label"
        class="Select-module_menu__1H2aU"
        id="accessibilityProfile-menu"
        role="listbox"
        style="max-height: 260px;"
        tabindex="-1"
      />
    </div>
  </div>
</div>
`;
